src/G2/Equiv/EquivADT.hs

```diff
@@ -123,4 +125,7 @@ exprPairing ns s1@(State {expr_env = h1}) s2@(State {expr_env = h2}) e1 e2 pairs
     (Type _, Type _) -> Just pairs
     (Case _ _ _, _) -> Just (HS.insert (Ob e1 e2) pairs)
     (_, Case _ _ _) -> Just (HS.insert (Ob e1 e2) pairs)
+    -- TODO not sure if I should have it this way
+    (Var _, _) -> Just (HS.insert (Ob e1 e2) pairs)
+    (_, Var _) -> Just (HS.insert (Ob e1 e2) pairs)
```

Comment:
What's the goal here?  Were some cases with variables falling through to the error at line 131?  More broadly, maybe line 131 should just insert an Ob into the hashset, and then we can get rid of all these special cases for Lambdas, Cases, etc?  Or is that a bad idea for some reason that I am not seeing?

Reply:
Yes, there was one rule that I noticed was failing before:  infEq in Zeno.hs.